docs/language/pipelines.md

```diff
@@ -169,7 +169,17 @@ Configuration Parameters:
 
 - `interval`: Time interval in nanoseconds after which the window closes.
 
+##### Windows Based on State
 
+The most flexible windows are code or state based windows. Instead of being fixed on the timestamp
+or a message cout it allows to express the window logic of closing and emitting in tremor-script.
+
+This requires two parts:
+
+1) the `state` section of the window defines the inital state, this is the same as `state` for the script operator.
+2) the `script from tick` section defines the logic that is executed on the periodic ticks to ensure windows that don't get new events can be closed and emitted.
```

Comment:
```suggestion
2) the `script from tick` section defines the logic that is executed on periodic ticks to ensure windows that don't get new events can be closed and emitted.
```

We need to document what the tick interval is. Is it still 100ms?

Reply:
We need to add that the normal `script` is executed for each incoming event, and there `event` and `$` are available, no?

Reply:
We also need to document that for ticks, the current tick time can be fetched by using `tremor::system::ingest_ns()`.

Reply:
ja good call!